x/erc20/keeper/ibc_callbacks.go

```diff
@@ -82,39 +67,61 @@ func (k Keeper) OnRecvPacket(
 		data.Denom, data.Amount,
 	)
 
-	// check if the coin is a native staking token
-	bondDenom := k.stakingKeeper.BondDenom(ctx)
-	if coin.Denom == bondDenom {
-		// no-op, received coin is the staking denomination
-		return ack
-	}
-
 	pairID := k.GetTokenPairID(ctx, coin.Denom)
-	if len(pairID) == 0 {
-		// short-circuit: if the denom is not registered, conversion will fail
-		// so we can continue with the rest of the stack
+	pair, found := k.GetTokenPair(ctx, pairID)
+
+	// TODO: Consider how it integrates with PFM.
+	// Case 1 - token pair is not registered
+	// Case 1.1 - coin is a native chain voucher and the token pair is not registered
+	if !found && ibc.IsSingleHop(coin.Denom) {
+		denomAddr, err := utils.GetIBCDenomAddress(coin.Denom)
+		if err != nil {
+			return channeltypes.NewErrorAcknowledgement(err)
+		}
+
+		// TODO: Where are we populating the name, symbol and decimals for the metadata ?
+		// For an IBC voucher will there be metadata already available in the bank ?
+		// coinMetadata, found := k.bankKeeper.GetDenomMetaData(ctx, coin.Denom)
+		coinMetadata, err := k.CreateERC20Metadata(ctx, coin.Denom, "", "", 0)
+		if err != nil {
+			return channeltypes.NewErrorAcknowledgement(err)
+		}
+
+		// Truncate the base denom to 20 characters
+		truncatedAddr := denomAddr[:20]
+		coinMetadata.Base = common.BytesToAddress(truncatedAddr).String()
+
+		tokenPair, err := k.RegisterTokenPairForNativeCoin(ctx, *coinMetadata)
+		if err != nil {
+			return channeltypes.NewErrorAcknowledgement(err)
+		}
+
+		if err := k.RegisterPrecompileForCoin(ctx, *tokenPair); err != nil {
+			return channeltypes.NewErrorAcknowledgement(err)
+		}
+
 		return ack
 	}
 
-	pair, _ := k.GetTokenPair(ctx, pairID)
-	if !pair.Enabled {
-		// no-op: continue with the rest of the stack without conversion
+	// Case 2 - Coin is native EVMOS
+	if pair.Denom == utils.BaseDenom {
+		// no-op: continue with the rest of the stack
 		return ack
 	}
```

Comment:
Is this case relevant, would be ideal to return ack earlier so it doesn't check further?